### PR TITLE
Upgrades React and several dependencies to fix security vulnerabilities.

### PR DESCRIPTION
**Pull request by maritafarruggia**
**What**
- React v.16.9.0 —> 16.12.0
- React-dom v. 16.9.0 —> 16.12.0
- React-scripts v. 1.1.4 —> 3.3.1
- React-router v.4.2.0 —> 5.1.2
- React-router-dom v. 4.2.2 —> 5.1.2
- Enzyme v.3.3.0 —> 3.11.0

- Upgraded set-value, mixin-deep, js-yaml, extend, react-dev-utils, webpack-dev-server, url-parse, lodash, lodash,template & mem.

**Why**
Fixes security vulnerabilities.

### CODE REVIEW

**Comment by maritafarruggia** [comment]
[Upgrade React Code Extension to latest React version](https://trello.com/c/7vYBudLJ/776-upgrade-react-code-extension-to-latest-react-version)

File: web_full_stack/js/.node-version [hunk 1]
```diff
@@ -1 +0,0 @@
-10.13.0
```
Comment (blevine-reverb) [inline web_full_stack/js/.node-version]: What node version should we be running now?

File: web_full_stack/js/README.md [hunk 2]
```diff
@@ -2,13 +2,13 @@ JavaScript Code Extension
 ---
 The JS Code Extension is a React application created using [create-react-app](https://github.com/facebook/create-react-app). Tests are written using [enzyme](https://github.com/airbnb/enzyme) and [react-router](https://github.com/ReactTraining/react-router) manages routing.
 
+# Setup
 You'll need [yarn](https://yarnpkg.com/en/) installed and a recent version (>= 8) of [Node](https://nodejs.org/en/) to use this project. If you're using homebrew you can brew install both of those or if you want to manage multiple versions of node on your machine, we suggest using a tool like [nodenv](https://github.com/nodenv/nodenv).
```
Comment (blevine-reverb) [inline web_full_stack/js/README.md]: Let's double check that this is still true. Might need a newer version with all the updates

File: web_full_stack/js/README.md [hunk 3]
```diff
@@ -2,13 +2,13 @@ JavaScript Code Extension
...
 
-# Setup
-* `yarn install`
+* `yarn install` to install dependencies.
 
-## Running/Tests
-* `yarn run start` to start the development server.
+# Running/Tests
+* `yarn start` to start the development server. Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
 * `yarn run test` to start the test suite.
```
Comment (blevine-reverb) [inline web_full_stack/js/README.md]: You can get rid of `run` here too and make it just `yarn test`.

**Comment by blevine-reverb** [comment]
>New Javascript app folder replaces old JS one.

Is that still true? I don't see any of those changes in the diff, though maybe github is hiding renamed files.

**Comment by maritafarruggia** [comment]
> > New Javascript app folder replaces old JS one.
> 
> Is that still true? I don't see any of those changes in the diff, though maybe github is hiding renamed files.

Github is hiding the renames.

File: web_full_stack/js/public/manifest.json [hunk 4]
```diff
@@ -6,9 +6,19 @@
...
+      "type": "image/png",
+      "sizes": "192x192"
+    },
+    {
+      "src": "logo512.png",
+      "type": "image/png",
+      "sizes": "512x512"
     }
   ],
-  "start_url": "./index.html",
```
Comment (davidmackhello) [inline web_full_stack/js/public/manifest.json]: Do we no longer need to specify `index.html` as the entrypoint?

**Pull request merged**